Makefile

```diff
@@ -20,36 +21,53 @@ $(info !!!!! Defaulting to Linux x86_64 build options !!!!!)
 BUILD_OPT_EXE = $(BUILD_OPT_EXE.Linux.x86_64)
 endif
 
-pnut-sh: pnut.c sh.c sh-runtime.c
-	mkdir -p $(BUILD_DIR)
-	gcc $(BUILD_OPT_SH) pnut.c -o $(BUILD_DIR)/pnut-sh
+.PHONY: default install uninstall clean test-sh test-i386-linux test-x86_64-linux test-x86_64-mac
+
+default: $(BUILD_DIR)/pnut-sh $(BUILD_DIR)/pnut-exe
+
+all: $(BUILD_DIR)/pnut-sh \
+	$(BUILD_DIR)/pnut-sh.sh \
+	$(BUILD_DIR)/pnut-sh-bootstrapped.sh \
+	$(BUILD_DIR)/pnut-exe \
+	$(BUILD_DIR)/pnut-exe.sh \
+	$(BUILD_DIR)/pnut-exe-bootstrapped
+
+$(BUILD_DIR)/config.mk:
+	./configure
+
+include $(BUILD_DIR)/config.mk
+
+PNUT_SH_DEPS=pnut.c debug.c sh.c sh-runtime.c env.c
+PNUT_EXE_DEPS= pnut.c debug.c x86.c exe.c env.c elf.c mach-o.c
```

Comment:
I didn't know about `gcc -MM`. It would have been useful when [implementing the `measure-file-size.sh` script](https://github.com/udem-dlteam/pnut/blob/master/analysis/measure-file-size.sh#L43-L50)!